modules/homebrew.nix

```
@@ -1,219 +1,780 @@
+
+    config = {
+      brewBundleCmd = concatStringsSep " " (
+        optional (!config.autoUpdate) "HOMEBREW_NO_AUTO_UPDATE=1"
+        ++ [ "brew bundle --file='${brewfileFile}' --no-lock" ]
+        ++ optional (!config.upgrade) "--no-upgrade"
+        ++ optional (config.cleanup == "uninstall") "--cleanup"
+        ++ optional (config.cleanup == "zap") "--cleanup --zap"
+      );
+    };
+  };
+
+  globalOptions = { config, ... }: {
+    options = {
+      brewfile = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable Homebrew to automatically use the Brewfile that this module generates in
+          the Nix store, when you manually invoke <command>brew bundle</command>.
+
+          Enabling this option will change the default value of
+          ${mkDocOptionLink "homebrew.global.lockfiles"} to <literal>false</literal> since, with
+          this option enabled, <command>brew bundle [install]</command> will default to using the
+          Brewline that this module generates in the Nix store, unless you explicitly point it at
+          another Brewfile using the <literal>--file</literal> flag. As a result, it will try to
+          write the lockfile in the Nix store, and complain that it can't (though the command will
+          run successfully regardless).
+
+          Implementation note: when enabled, this option sets the
+          <literal>HOMEBREW_BUNDLE_FILE</literal> environment variable to the path of the Brewfile
+          that this module generates in the Nix store, by adding it to
+          ${mkDocOptionLink "environment.variables"}.
+        '';
+      };
+      autoUpdate = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to enable Homebrew to auto-update itself and all formulae when you manually invoke
+          commands like <command>brew install</command>, <command>brew upgrade</command>,
+          <command>brew tap</command>, and <command>brew bundle [install]</command>.
+
+          Note that Homebrew auto-updates when you manually invoke commands like the ones mentioned
+          above if it's been more then 5 minutes since it last updated.
+
+          You may want to consider disabling this option if you have
+          ${mkDocOptionLink "homebrew.onActivation.upgrade"} enabled, and
+          ${mkDocOptionLink "homebrew.onActivation.autoUpdate"} disabled, if you want to ensure that
+          your installed formulae will only be upgraded during <command>nix-darwin</command> system
+          activation, after you've explicitly run <command>brew update</command>.
+
+          Implementation note: when disabled, this option sets the
+          <literal>HOMEBREW_NO_AUTO_UPDATE</literal> environment variable, by adding it to
+          ${mkDocOptionLink "environment.variables"}.
+        '';
+      };
+      lockfiles = mkOption {
+        type = types.bool;
+        default = true;
```

Maybe we should make clear that the default depends on the brewfile option?
```suggestion
        default = !brewfile;
```

---

Yeah, that does seem better. I had considered doing this, but didn't since I was under the impression for some reason that using values from `config` when setting option defaults would cause infinite recursion issues.

I've now changed this to,
```
        default = !config.brewfile;
        defaultText = literalExpression "!config.homebrew.global.brewfile";
```

and removed,
```
      lockfiles = mkIf config.brewfile (mkDefault true);
```